File: angular/src/app/service/model/project-invoice-setting.dto.ts
export interface ProjectInvoiceSettingDto {
    currencyName: string,
    invoiceNumber: number,
    discount: number,
    isMainProjectInvoice: boolean,
    mainProjectId?: number,
    subProject: IdNameDto[],
    subProjectNames: string[],
    subProjectIds: number[],
    otTypes?: { id?: number, otTypeName: string, multiplier: number, isNormalInvoice: boolean }[],
}

export interface IdNameDto {
    id: number,
    name: string
}

File: aspnet-core/src/ProjectManagement.Application/APIs/ProjectUserBills/Dto/ProjectInvoiceSettingDto.cs
using ProjectManagement.APIs.Timesheets.Dto;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ProjectManagement.APIs.ProjectUserBills.Dto
{
    public class ProjectInvoiceSettingDto
    {
        public string CurrencyName { get; set; }        
        public long InvoiceNumber { get; set; }
        public float Discount { get; set; }

        public bool IsMainProjectInvoice { get; set; } = true;
        public long? MainProjectId { get; set; }
        public string MainProjectName { get; set; }
        public List<IdNameDto> SubProjects { get; set; }
        public List<string> SubProjectNames => SubProjects.Select(s => s.Name).ToList();
        public List<long> SubProjectIds => SubProjects.Select(s => s.Id).ToList();
        public List<OtTypeDto> OtTypes { get; set; }
    }

    public class OtTypeDto
    {
        public long? Id { get; set; }
        public string OtTypeName { get; set; }
        public float Multiplier { get; set; }
    }

}
